api/client/client.go

```diff
@@ -145,22 +145,47 @@ func (srv *Server) Sign(jsonData []byte) ([]byte, error) {
 }
 
 // Info sends an info request to the remote CFSSL server, receiving a
-// certificate or an error in response.
+// response or an error in response.
 // It takes the serialized JSON request to send.
-func (srv *Server) Info(jsonData []byte) ([]byte, error) {
-	return srv.Req(jsonData, "info")
+func (srv *Server) Info(jsonData []byte) (*InfoResp, error) {
+	res, err := srv.getResultMap(jsonData, "info")
+	if err != nil {
+		return nil, err
+	}
+
+	cert := res["certificate"]
+	usages := res["usages"].([]interface{})
+	exp := res["expiry"]
+
+	usageStrings := make([]string, len(usages), len(usages))
+	for i, s := range usages {
+		usageStrings[i] = s.(string)
+	}
+
+	return &InfoResp{
+		Certificate:  cert.(string),
+		Usage:        usageStrings,
+		ExpiryString: exp.(string),
+	}, nil
+}
+
+func (srv *Server) getResultMap(jsonData []byte, target string) (result map[string]interface{}, err error) {
+	url := srv.getURL(target)
+	response, err := srv.post(url, jsonData)
+	if err != nil {
+		return
+	}
+	result = response.Result.(map[string]interface{})
```

Comment:
A few things here:

* You should check whether response.Result is nil
* You should make sure it can be coerced to a `map[string]interface{}`. You can do that with something like:

```
result, ok := response.Result.(map[string]interface{})
if !ok {
        // return an error
}
return result, nil
```